app/controllers/badges_controller.rb

```diff
@@ -49,11 +49,15 @@ def destroy
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_badge
-      @badge = policy_scope(Badge).find(params[:id])
+      @badge = authorize policy_scope(Badge).find(params[:id])
     end
 
     # Only allow a list of trusted parameters through.
     def badge_params
-      params.require(:badge).permit(:name, :description, :color, :short, :badge_type_id, :team_id, person_ids: [])
+      if policy(@badge).update?
```

Comment:
Creating a new badge here shoots an error.

 "unable to find policy `NilClassPolicy` for `nil`"